iree/base/BUILD

```diff
@@ -460,10 +465,13 @@ cc_library(
     copts = [
         "-D_GNU_SOURCE=1",
     ],
-    linkopts = [
-        "-ldl",
-        "-lpthread",
```

Comment:
BTW, other uses of this are around
https://github.com/google/iree/blob/584e61c87303fd1108e0cc732e89deff0ea76c1d/build_tools/bazel/third_party_import/llvm-project/overlay/mlir/BUILD.bazel#L3363-L3366

which could maybe use 
https://github.com/tensorflow/tensorflow/blob/9a36023aa3d6d50a77693da2faa9095d6c8ace8e/third_party/llvm/llvm.bzl#L355-L359
like LLVM